tests/unit/test-cmd/test.run.js

```diff
@@ -234,6 +234,14 @@ describe('run', () => {
     sinon.assert.calledOnce(desktopRunnerStub);
   });
 
+  it('will not run extension runner if target is not supported', async () => {
+    const cmd = prepareRun();
+    await cmd.run({target: ['not-supported']});
+
+    sinon.assert.notCalled(androidRunnerStub);
+    sinon.assert.notCalled(desktopRunnerStub);
+  });
+
```

Comment:
This test doesn't help much, because this tests the internal implementation, and not the CLI frontend. If you add `sinon.assert.notCalled(cmd.options.MultiExtensionRunner)`, then you would see that the test fails, because the runner is run anyway.

There are two relevant conditions to check:
1. `-t firefox-desktop` works.
2. `-t firefox-desktop -t not-supported -t firefox-desktop` results in an error message.

To test (1), you can add the parameter after this line: https://github.com/mozilla/web-ext/blob/72bb28ca8e29e18129fc24bb26c85d9ce5f10b82/tests/functional/test.cli.run.js#L20

To test (2), you could copy that test, cut the parts that we don't need, and add it to the same file.

https://github.com/mozilla/web-ext/blob/master/tests/functional/test.cli.run.js#L11

Reply:
Would it be fine if I use `assert` to validate the `exitCode` and `stderr`?

Reply:
Yes, that works. The `reportCommandErrors` in the existing test mainly exists to help with debugging if there is an unexpected test failure.